### PR TITLE
prevent duplicate words

### PR DESCRIPTION
**Pull request by gaoDean**
self-explanatory; lol i dont know anything about golang, if u find anything like bad practices pls tell me; I am also bad at github and I don't know how to separate commits into branches, so I just committed the last pull request with this one, pls tell me how to do properly

### CODE REVIEW

File: src/util.go [hunk 1]
```diff
@@ -80,11 +80,23 @@ func init() {
 	rand.Seed(time.Now().Unix())
 }
 
+func lastField(r string) string {
+    ss := strings.Fields(r) // seperated string
+    return ss[len(ss) - 1] // get last word
```
Comment (rvaiya) [inline src/util.go:85]: You should always make sure len(ss) != 0 when writing a generic function like this. In this case we know the caller will always present an r with at least one word, but in general you should make no assumptions.

File: src/util.go [hunk 2]
```diff
@@ -80,11 +80,23 @@ func init() {
...
 func randomText(n int, words []string) string {
 	r := ""
 
-	for i := 0; i < n; i++ {
-		r += words[rand.Int()%len(words)]
+    r += words[rand.Int()%len(words)]
+    r += " "
+    for i := 0; i < n; i++ {
+        new := words[rand.Int()%len(words)]
+        for lastField(r) == new {
```
Comment (rvaiya) [inline src/util.go:95]: Instead of extracting the last word from `r` (inefficient) it would be better to store the last word and compare it explicitly.

**Review by rvaiya — COMMENTED** [review]
Sorry for the delay. 

Additional Note:

You should always run your code through gofmt. You've used spaces for indentation.

File: src/util.go (hunk 1 above)
Comment (gaoDean) [reply src/util.go:85]: Sorry, I am quite new to golang, how would I make sure that it is more than 0? Is there like some kind of exit function that stops the process?

**Comment by gaoDean** [comment]
alright, this should be good to go? Have removed the lastfield function and implemented your way of storing last word @rvaiya

**Comment by lemnos** [comment]
Sorry for the abysmal response time. This looks good and has been merged in (5a179f25). I squashed the commits into a single commit (you should do this when submitting PRs, since it keeps the history nice and clean). One minor quibble is that 'new' is a reserved keyword in go. While the code compiles, it is probably better to avoid this, so I renamed the variable in 65af3c34.

**Comment by gaoDean** [comment]
Oh nice, i totally forgot about this.

**Pull request closed without merging**